services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/RouteOptions.java

```diff
@@ -109,28 +118,76 @@ public static Builder builder() {
+   * Values can be any number greater than 0 or the string unlimited. A NoSegment error is returned
+   * if no routable road is found within the radius.
    *
-   * @return a string representing the radiuses
+   * @return a string representing the radiuses separated by ;.
    * @since 3.0.0
+   * @deprecated use {@link #radiusesList()} ()}
+   */
+  @Nullable
+  @Deprecated
+  public String radiuses() {
+    return FormatUtils.formatRadiuses(radiusesList());
+  }
+
+  /**
+   * The maximum distance a coordinate can be moved to snap to the road network in meters. There
+   * must be as many radiuses as there are coordinates in the request.
+   * Values can be any number greater than 0 or the string unlimited. A NoSegment error is returned
+   * if no routable road is found within the radius.
+   *
+   * @return a list of radiuses
    */
   @Nullable
-  public abstract String radiuses();
+  public abstract List<Double> radiusesList();
 
   /**
-   * The same bearings which were used during the request that resulted in this root directions
-   * response. Note that even though these are saved. it's a good idea to recalculate any bearings
-   * being used and use the newer values for the directions request.
+   * Influences the direction in which a route starts from a waypoint. Used to filter the road
+   * segment the waypoint will be placed on by direction. This is useful for making sure the new
+   * routes of rerouted vehicles continue traveling in their current direction. A request that does
+   * this would provide bearing and radius values for the first waypoint and leave the remaining
+   * values empty. Takes two comma-separated values per waypoint: an angle clockwise from true north
+   * between 0 and 360, and the range of degrees by which the angle can deviate (recommended value
+   * is 45° or 90°), formatted as {angle, degrees}. If provided, the list of bearings must be the
+   * same length as the list of coordinates. However, you can skip a coordinate and show its
+   * position in the list with the ; separator.
    *
-   * @return a string representing the bearings used in the original request
+   * @return a string representing the bearings with the ; separator. Angle and degrees for every
+   *   bearing value are comma-separated.
    * @since 3.0.0
+   * @deprecated use {@link #bearingsList()}
+   */
+  @Nullable
+  @Deprecated
+  public String bearings() {
+    return FormatUtils.formatBearings(bearingsList());
+  }
+
+  /**
+   * Influences the direction in which a route starts from a waypoint. Used to filter the road
+   * segment the waypoint will be placed on by direction. This is useful for making sure the new
+   * routes of rerouted vehicles continue traveling in their current direction. A request that does
+   * this would provide bearing and radius values for the first waypoint and leave the remaining
+   * values empty. Takes two comma-separated values per waypoint: an angle clockwise from true north
```

Comment:
This does not take, but return a value that's a list of double.

Reply:
fixed